reward/addressBookManager.go

```diff
@@ -0,0 +1,186 @@
+package reward
+
+import (
+	"errors"
+	"fmt"
+	"github.com/klaytn/klaytn/accounts/abi"
+	"github.com/klaytn/klaytn/blockchain"
+	"github.com/klaytn/klaytn/blockchain/types"
+	"github.com/klaytn/klaytn/blockchain/vm"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/contracts/reward/contract"
+	"github.com/klaytn/klaytn/params"
+	"math/big"
+	"strings"
+)
+
+// addressType defined in AddressBook
+const (
+	addressTypeNodeID = iota
+	addressTypeStakingAddr
+	addressTypeRewardAddr
+	addressTypePoCAddr
+	addressTypeKIRAddr
+)
+
+var (
+	errAddressBookIncomplete = errors.New("incomplete node information from AddressBook")
+)
+
+type addressBookManager struct {
+	bc                         *blockchain.BlockChain
+	chainConfig                *params.ChainConfig
```

Comment:
I'm not sure this chainConfig field is necessary.

Reply:
reflect